srcCpp/perftest_publisher.cxx

```diff
@@ -1850,12 +1867,27 @@ int perftest_cpp::Publisher()
                     // back the LENGTH_CHANGED_SIZE message
                     message.latency_ping = num_pings % _NumSubscribers;
 
-                    while (announcement_reader_listener->announced_subscribers > 0) {
+                    /*
+                     * If the Throughput topic is reliable, we can send the packet and do
+                     * a wait for acknowledgements. However, if the Throughput topic is
+                     * Best Effort, wait_for_acknowledgments() will return inmediately.
+                     * This would cause that the Send() would be exercised too many times,
+                     * in some cases causing the network to be flooded, a lot of packets being
+                     * lost, and potentially CPU starbation for other processes.
+                     * We can prevent this by adding a small sleep() if the test is best
+                     * effort.
+                     */
+
+                    while (announcement_reader_listener->announced_subscriber_replies > 0) {
                         writer->Send(message, true);
                         writer->Flush();
-                        writer->wait_for_acknowledgments(
+                        if (_IsReliable) {
```

Comment:
Instead of implementing this distinction here, I would move the logic inside wait_for_acknowledgements. perftest_publisher is supposed to be middleware agnostic and here you are making assumptions based on reliability. In addition, it will save the If in multiple places

Reply:
I agree that would be a saving, however it would require to add to the RTIPublisher class a variable to check if the publisher is reliable or not (since, currently we don't have access to that information, do you want to proceed that way? @fercsrti77